hall_calc_example.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+from haystack.dataclasses import ChatMessage
+
+from haystack_experimental.components.generators.chat.openai import HallucinationScoreConfig, OpenAIChatGenerator
+
+os.environ["OPENAI_API_KEY"] = ""
+
+llm = OpenAIChatGenerator(model="gpt-4o")
+
+# Closed Book Example
+closed_book_result = llm.run(
+    messages=[ChatMessage.from_user(text="Who won the 2019 Nobel Prize in Physics?")],
+    hallucination_score_config=HallucinationScoreConfig(skeleton_policy="closed_book"),
+)
+print(f"Decision: {closed_book_result['replies'][0].meta['hallucination_decision']}")
+print(f"Risk bound: {closed_book_result['replies'][0].meta['hallucination_risk']:.3f}")
+print(f"Risk bound: {closed_book_result['replies'][0].meta['hallucination_rationale']}")
+print(f"Answer:\n{closed_book_result['replies'][0].text}")
+print("---\n")
+
+# Evidence-based Example
```

Comment:
Would be great to have also a RAG Pipeline example that uses a DocumentStore and a Retriever. It doesn't stop us from merging this PR but I believe it would help users to try out the new feature and we should add it in another PR.